src/wpool_worker.erl

```diff
@@ -18,6 +18,12 @@
 
 -behaviour(gen_server).
 
+-ifdef(OTP_RELEASE).
```

Comment:
I still won't merge a PR with macros and this kind of backward compatibility, sorry.
It's just how we manage our projects at @inaka.
As I told @michalwski, if our users need to maintain older versions of Erlang then we can backport patches and create releases for old versions + new patches, but if we move ahead to OTP21, we move entirely.